src/user/model.rs

```diff
@@ -36,40 +39,55 @@ pub struct UserData {
 
 #[derive(Debug, Serialize, Deserialize, Clone)]
 pub struct SlimUser {
-    pub email: Option<String>,
+    pub user_uuid: Uuid,
+    pub email: String,
+    pub role: String,
 }
 
-#[derive(Debug, Deserialize)]
-pub struct AuthData {
-    pub email: String,
-    pub password: String,
+#[derive(Shrinkwrap, Clone, Default)]
+pub struct LoggedUser(pub Option<SlimUser>);
```

Comment:
I think that `Shrinkwrap` (in fact `newtypes`) are really amazing for wrapping these stuff, and ensuring that the code only sees the stuff it needs to see and not mix things up.